sources/core/Stride.Core.Mathematics.Tests/TestMatrix.cs

```diff
@@ -0,0 +1,109 @@
+
+            // Yaw-Pitch-Roll is the intrinsic rotation order, so extrinsic is the reverse (ie. Z-X-Y)
+            var rotMatrix = Matrix.RotationZ(rollRadians) * Matrix.RotationX(pitchRadians) * Matrix.RotationY(yawRadians);
+            rotMatrix.Decompose(out float decomposedYaw, out float decomposedPitch, out float decomposedRoll);
+
+            var expectedQuat = Quaternion.RotationYawPitchRoll(yawRadians, pitchRadians, rollRadians);
+            var decompedQuat = Quaternion.RotationYawPitchRoll(decomposedYaw, decomposedPitch, decomposedRoll);
+            Assert.True(expectedQuat == decompedQuat || expectedQuat == -decompedQuat, $"Quat not equals: Expected: {expectedQuat} - Actual: {decompedQuat}");
+        }
+
+        [Theory, MemberData(nameof(RotationAngles))]
+        public void TestDecomposeXYZFromMatricesXYZ(float yawDegrees, float pitchDegrees, float rollDegrees)
+        {
+            var yawRadians = MathUtil.DegreesToRadians(yawDegrees);
+            var pitchRadians = MathUtil.DegreesToRadians(pitchDegrees);
+            var rollRadians = MathUtil.DegreesToRadians(rollDegrees);
+
+            var rotMatrix = Matrix.RotationX(pitchRadians) * Matrix.RotationY(yawRadians) * Matrix.RotationZ(rollRadians);
+            rotMatrix.DecomposeXYZ(out Vector3 eulerAngles);
+
+            var decompedRotMatrix = Matrix.RotationX(eulerAngles.X) * Matrix.RotationY(eulerAngles.Y) * Matrix.RotationZ(eulerAngles.Z);
+            var decompedQuat = Quaternion.RotationMatrix(decompedRotMatrix);
+
+            var expectedQuat = Quaternion.RotationX(pitchRadians) * Quaternion.RotationY(yawRadians) * Quaternion.RotationZ(rollRadians);
+            Assert.True(expectedQuat == decompedQuat || expectedQuat == -decompedQuat, $"Quat not equals: Expected: {expectedQuat} - Actual: {decompedQuat}");
+        }
+
+        public static IEnumerable<object[]> RotationAngles
+        {
+            get
+            {
+                var result = new List<object[]>();
+                int angleIncrementDegrees = 15;
+                int angleTestCount = 180 / angleIncrementDegrees;       // Up to 180 degrees in postive & negative direction should be enough to test
+                for (int yawIdx = 0; yawIdx <= angleTestCount; yawIdx++)
+                {
+                    for (int pitchIdx = 0; pitchIdx <= angleTestCount; pitchIdx++)
+                    {
+                        for (int rollIdx = 0; rollIdx <= angleTestCount; rollIdx++)
+                        {
+                            float yawDegrees = yawIdx * angleIncrementDegrees;
+                            float pitchDegrees = pitchIdx * angleIncrementDegrees;
+                            float rollDegrees = rollIdx * angleIncrementDegrees;
+
+                            result.Add(new object[] { yawDegrees, pitchDegrees, rollDegrees });
+                            if (yawDegrees != 0 && pitchDegrees != 0 && rollDegrees != 0)
```

Comment:
Why are we only adding negative case if all three variables are positive? It's not testing (0, 0, -15) for example.

Reply:
My mistake, have changed this now (see the other comment).